scanpy/tools/_louvain.py

```diff
@@ -133,7 +136,24 @@ def louvain(
                 g, partition_type,
                 **partition_kwargs,
             )
-            # adata.uns['louvain_quality'] = part.quality()
+            if 'quality' in dir(part):
+                q = part.quality()
+                if isinstance(part, louvain.RBConfigurationVertexPartition) and q > 1.0: #scale
+                    m = weights.sum() if use_weights else g.ecount()
+                    q /= m if directed else m*2
+                if isinstance(part, (
+                    louvain.RBConfigurationVertexPartition,
+                    louvain.ModularityVertexPartition,
+                )):
+                    qual_type = ' (scaled modularity)'
+                else:
+                    qual_type = ''
+                adata.uns[uns_key]['quality'] = q
+                quality_msg = (
+                    f'\n'
+                    f'    quality of the partitioning{qual_type} is {q:.3f}\n'
+                    f'    added "quality" key to adata.uns["{uns_key}"]'
```

Comment:
Should the `partition_type`, `random_state`, and `resolution` be logged as well?

Reply:
I'm adding resolution, too. But I feel like adding all of them would be too verbose.

Reply:
Fair enough. I just like the idea of being able to run the exact same call by using the parameters.